doc/tutorials/data_loading/datasetmapper.txt

```diff
@@ -0,0 +1,178 @@
+/*!
+
+@file datasetmapper.txt
+@author Gopi Tatiraju
+@breif Introduction and tutorial for how to use DatasetMapper in mlpack.
+
+@page datasetmapper DatasetMapper Tutorial 
+
+@section intro_datasetmapper Introduction
+
+DatasetMapper is a class which holds information about a dataset. This can be
+used when dataset contains categorical non-numeric features which should be
+mapped to numeric features. A simple example can be
+
+```
+7,5,True,3
+6,3,False,4
+4,8,False,2
+9,3,True,3
+```
+
+The above dataset will be respresented as
+
+```
+7,5,0,3
+6,3,1,4
+4,8,1,2
+9,3,0,3
+```
+
+Here Mappings are
+
+- True mapped to 0
+- False mapped to 1
```

Comment:
@rcurtin @heisenbuug I may be missing the context here, but in boolean logic isn't `False` usually mapped to `0` and True` mapped to `1` ?

Reply:
@conradsnicta good point!  Actually the code @heisenbuug wrote is correct---basically, our categorical mapping will map the first non-numeric value it encounters to `0`, and in the example, `True` is seen first.  But, you are right, that is a little bit misleading.

@heisenbuug do you think we can change the example so that the categorical values have `False` first then `True` second? (You could just switch them in the input CSV.)  Alternately, you could choose a different example than true/false to take away the ambiguity.

Reply:
@conradsnicta @rcurtin I actually was confused when saw the output but then when went through the comments in the code then realized that it will start mapping with 0.

So I kept the example in this way so that when user will face this issue while using the `DatasetMapper` they will realize it as well but I think should have added a comment about it here.

Should I add a comment or just change the example? Cause if we change it, there is a chance that in future someone might ask about the same in an issue. Let me know.

Also, I will push the suggested changes soon, sorry for the delay.

Reply:
Ah, yeah, that is another idea---if you add a note to clearly point out that we convert non-numeric values to integers in the order that we encounter them, that should solve the possible confusion too.